
canyon_mapper_output:
  cloud_list:           [nav_cloud, map_cloud, local_dense_cloud]
# This cloud is used for navigation, updated regularly and kept sparse
  nav_cloud: 
    desired_frame:              base_link
    # Publishing
    publish:                    true
    incremental_update:         false
    throttle_publish:           false
    publishing_throttle_max:    1
    retain_after_scan:          false
    should_save:                false
    # Processing Settings
    # NOTE - task names in task_list below MUST match the subheadings under the tasks heading
    preprocessing:
      should_process:             true
      min_cloud_size:             50
      throttle:                   false
      throttle_max:               25
      task_list:            [ voxelize, clip ]
      tasks:
        transform:
          type:             1     # Transformation
          map_name:         map
          should_publish:   true
          publish_topic:    transformed_cloud
        clip: 
          type:             2     # Clipping 
          box:              [ -5, 5, -3, 3, -1, 3.7, 0,0,0,0,0,0 ]
          keep_ordered:     false
          should_publish:   true
          publish_topic:    clipped_cloud
        voxelize:           
          type:             3     # Voxelization
          leaf_sizes:       [ 0.03, 0.03, 0.03 ]
          keep_ordered:     false
          should_publish:   true
          publish_topic:    voxelized_cloud
    postprocessing: 
      should_process:       false
# This is the dense map cloud, updated only sporadically
  map_cloud: 
    desired_frame:              map
    # Publishing
    publish:                    true
    incremental_update:         true
    throttle_publish:           false
    publishing_throttle_max:    50
    retain_after_scan:          true
    should_save:                false
    # Processing Settings
    # NOTE - task names in task_list below MUST match the subheadings under the tasks heading
    preprocessing:
      should_process:             true
      min_cloud_size:             50
      throttle:                   false
      throttle_max:               100
      task_list:            [ voxelize, clip ]
      tasks:
        transform:
          type:             1     # Transformation
          map_name:         map
          should_publish:   true
          publish_topic:    transformed_cloud
        clip: 
          type:             2     # Clipping 
          box:              [ -20, 20, -20, 20, -1, 6, 0,0,0,0,0,0 ]
          keep_ordered:     false
          should_publish:   true
          publish_topic:    clipped_cloud
        voxelize:           
          type:             3     # Voxelization
          leaf_sizes:       [ 0.02, 0.02, 0.02 ]
          keep_ordered:     false
          should_publish:   true
          publish_topic:    voxelized_cloud
    postprocessing: 
      should_process:       false
# This is the last fully dense full scan 
  local_dense_cloud: 
    desired_frame:              base_link
    # Publishing
    publish:                    true
    incremental_update:         false
    throttle_publish:           false
    retain_after_scan:          false
    should_save:                true
    # Processing Data
    preprocessing: 
      should_process:       false
    postprocessing: 
      should_process:       false  